### PR TITLE
fixes #6289: avoids concurrent downloads of files with the same name and  destination dir.

### PR DESCRIPTION
**Pull request by dmitri-d**

### CODE REVIEW

**Comment by domcleal** [comment]
There seem to be some intermittent test failures, could you see if you can reproduce and debug?  It's a little concerning.

My only concern with this approach is that I don't think it'll work in a Passenger environment due to the class variables.  I don't know that it's a big problem though, since that's very much a minority deployment method.

**Comment by dmitri-d** [comment]
@domcleal: duh, one test was interfering with another.

File: lib/proxy/http_downloads.rb [hunk 1]
```diff
@@ -0,0 +1,45 @@
+module Proxy
+  class HttpDownloads
+    @@downloads_lock = Mutex.new
```
Comment (lzap) [inline lib/proxy/http_downloads.rb:3]: File lock might be better for those who run proxy in other stacks than webrick perhaps?

Finally :-) This was super annoying.

**Comment by lzap** [comment]
Closed intentionally?

Since we spawn a process, I think flock can replace mutex, but not big deal. LGTM

**Comment by dmitri-d** [comment]
@lzap, yeah, this one was superceded by https://github.com/theforeman/smart-proxy/pull/173

**Pull request closed without merging**